File: scripts/run_MR.py
#!/usr/bin/env python
'''
Created: 11/09
Updated: 11/14

Input: c10_per_rad_stm_kp.csv; run_MR.R
run_MR.R requires:
posterior_samples.savr; calc_mass_postpred.R

Output: KESPRINT-C10/*.png & .txt

Purpose: estimate mass of KESPRINT validated campaign 10
systems using MRrelation (Wolfgang 2015)

see also mass estimation of KESPRINT targets using M-R relation.ipynb

This script reads the list of validated planets (epic)
with their radius derived from transit observation.

The epic number, planet letter, radius, and radius error 
are fed as arguments to the run_MR.R script which produces 
the plots (.png) and posterior samples of mass prediction (.txt) 
The R script follows the example in github.
'''

import pandas as pd

#stellar parameters
df = pd.read_csv('c10_per_rad_stm.csv')

epics      = df['epic']
letter     = df['pl']
radii      = df['r']
radii_errs = df['r_err']

n=-1 #run all
for epic,let,r,r_err in zip(epics[:n],letter[:n],radii[:n],radii_errs[:n]):
    #print(epic)
    ###verbose
    !Rscript run_MR.R $epic $let $r $r_err
    ###use below to hide output in terminal
    !Rscript run_MR.R $epic $let $r $r_err 2&>1 >/dev/null

File: scripts/run_MR.R
#!/usr/bin/env Rscript

#setwd("~/github/MRrelation/")
scripts="~/github/MRrelation/scripts/"
data="~/github/MRrelation/data/"
outputdir="~/github/MRrelation/KESPRINT-C10/"

args <- commandArgs(trailingOnly = TRUE)

# test if there is at least one argument: if not, return an error
if (length(args)==0) {
  stop("(1)epic (2)planet_letter (3)planet_radius (4)planet_radius_err", call.=FALSE)
} else if (length(args)==1) {
  stop("(1)epic (2)planet_letter (3)planet_radius (4)planet_radius_err", call.=FALSE)
}

#parse arguments
epic    = args[1]
planletter = args[2]
planrad = as.double(args[3])
planraderr = as.double(args[4])

print(paste('------',epic,'------'))

#Generate samples from your planet radius distribution
numsamples = 100000 


#Read the posterior samples into R
load(paste(data,"posterior_samples.savr",sep=""))

#Read in the code that computes the mass posterior predictive distribution
source(paste(scripts,"calc_mass_postpred.R",sep=""))

print("---calculating posterior mass distribution---")

#Calculate the posterior predictive mass distribution
postpred = massguess_individpl(postsamp_eqn2_baseline,numsamples,planrad,raderr=planraderr)

masses = postpred$masses

print("Prediction: SUCESS")

#A plot showing the joint mass-radius distribution for this individual planet
#save plot as a png file
fname1 <- paste(epic,planletter,".png",sep="")
plotname = paste(outputdir,fname1,sep="")
png(file=plotname)
#postscript(file=plotfile,width=10,height=8)
plot_individMR(postpred$radii,postpred$masses)
#dev.copy(png, plotname)
dev.off()

fname2 <- paste("massrad_postpred_",epic,planletter,".txt",sep="")
#Write samples from joint mass-radius distribution into text file
write.table(postpred,file=paste(outputdir,fname2,sep=""))
